backend/src/user/service.ts

```diff
@@ -0,0 +1,86 @@
+import passport from "passport";
+import Strategy from "passport-google-oauth20";
+import { getCollections } from "../app";
+import { User } from "./model";
+import { Provider } from "./structures/Provider";
+
+type validUserOrError = [true, User] | [false, unknown];
+
+export function initializePassport() {
+    const GoogleStrategy = Strategy.Strategy;
+
+    passport.use(
+        new GoogleStrategy(
+            {
+                clientID: process.env.GOOGLE_CLIENT_ID as string,
+                clientSecret: process.env.GOOGLE_CLIENT_SECRET as string,
+                callbackURL: "http://localhost:8080/auth/google/callback",
+                userProfileURL: "https://www.googleapis.com/oauth2/v3/userinfo",
+                passReqToCallback: true,
+            },
+            async (_request, _accessToken, _refreshToken, profile, done: (arg0: null, arg1: any) => any) => {
+                let user = await getUserById(profile.id, "google");
```

Comment:
Zakładam, że implementacja tej funcji będzie niemal identyczna dla różnych strategii, także myślę, że warto to od razu wydzielić do osobnej funkcji, która będzie przyjmowała `(profile, provider)`, ewentualnie `(id, displayName, provider)`.